src/main/java/org/broadinstitute/consent/http/resources/DacResource.java

```diff
@@ -206,6 +213,36 @@ public Response filterUsers(@PathParam("term") String term) {
         return Response.ok().entity(users).build();
     }
 
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("{dacId}/dataset/{datasetId}")
+    @RolesAllowed({CHAIRPERSON})
+    public Response approveDataset(@Auth AuthUser authUser, @PathParam("dacId") Integer dacId, @PathParam("datasetId") Integer datasetId, String json) {
+        try{
+            User user = userService.findUserByEmail(authUser.getEmail());
+            Dataset dataset = datasetService.findDatasetById(datasetId);
+            if(dataset.getDacId() != dacId) { 
+                //Vague message is intentional, don't want to reveal too much info
+                throw new NotFoundException("Dataset not found");
+            }
+            user.checkIfUserHasRole(UserRoles.CHAIRPERSON.getRoleName(), dacId);
+            JsonObject payload = new Gson().fromJson(json, JsonObject.class);
```

Comment:
Another approach here is to create a very small model class (`DatasetApproval`?) specifically meant to capture the intended input argument. That would make it a tiny bit cleaner down below where we need to get the elements by string value and then coerce into a boolean.